### ISSUE TITLE
Adding a log exclude rule somehow excludes all JSON-parseable logs

### ISSUE THREAD

**Issue by mike-chen-samsung** [SumoLogic/sumologic-otel-collector#1128]
I installed Sumologic `3.7.0` Helm Chart in a `1.24` EKS cluster using Terraform with the value overrides:

```hcl
  set {
    name  = "sumologic.logs.container.otelcol.extraProcessors[0].filter/exclude_healthcheck_logs.logs.exclude.match_type"
    value = "regexp" 
  }
  set {
    name  = "sumologic.logs.container.otelcol.extraProcessors[0].filter/exclude_healthcheck_logs.logs.exclude.bodies[0]"
    value = "kube-probe" 
  }
```

And I verified in the Otelcol Logs configmap that the correct setting is being applied:
```
$ kubectl get cm  sumologic-sumologic-otelcol-logs  -o yaml
apiVersion: v1
data:
  config.yaml: |
    ...
    processors:
      filter/exclude_healthcheck_logs:
        logs:
          exclude:
            bodies:
            - kube-probe
            match_type: regexp
    ...
```

It correctly excludes logs containing the text `kube-probe`, and I do also see the expected logs show up in Sumologic, but only the ones that cannot be parsed as JSON. All JSON-parseable logs (that do not contain the excluded text) are somehow not making their way to Sumologic.

**Comment by swiatekm-sumo** [comment]
This may be a bug in the filter processor, where the [following condition](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/3f5370e4d1570a099efd64d9c7012a5d36309e5d/internal/filter/filterlog/filterlog.go#L119) doesn't return `false` for the match if the body isn't a string.

As a workaround, I'd suggest using OTTL directly:

```yaml
filter/exclude_healthcheck_logs:
  logs:
    error_mode: ignore
    logs:
      log_record:
        - 'IsMatch(body, ".*kube-probe*")'
```

**Comment by mike-chen-samsung** [comment]
Thanks for the suggestion. I tried it but ran into another error. Here is what I had:
```
$ kubectl get cm  sumologic-sumologic-otelcol-logs  -o yaml
apiVersion: v1
data:
  config.yaml: |
    ...
    processors:
      filter/exclude_healthcheck_logs:
        error_mode: ignore
        logs:
          log_record:
            - 'IsMatch(body, ".*kube-probe*")'
    ...
```

and here is the error:
```
Error: invalid configuration: processors::filter/exclude_healthcheck_logs: unable to parse OTTL statement: 1:44: unexpected token “<EOF>” (expected <opcomparison> Value)
2023/05/17 21:24:43 collector server run finished with error: invalid configuration: processors::filter/exclude_healthcheck_logs: unable to parse OTTL statement: 1:44: unexpected token “<EOF>” (expected <opcomparison> Value)
```

**Comment by swiatekm-sumo** [comment]
Try `IsMatch(body, ".*kube-probe*") == true` instead. This limitation got recently addressed upstream, but that change isn't yet present in the otel version 3.7.0 comes with. You'll be able to remove the `== true` part in 3.8.0.

**Comment by swiatekm-sumo** [comment]
This was fixed upstream in https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/22736 and released in `0.79.0-sumo-0`.
